code_climate.rb

```diff
@@ -0,0 +1,174 @@
+require "net/http"
+require "json"
+require "active_support/core_ext"
+
+DEFAULT_CODECLIMATE_URL = "https://api.codeclimate.com"
+
+class CodeClimateRequest
+  def initialize
+    @base_uri  = ENV["CODECLIMATE_URL"] || DEFAULT_CODECLIMATE_URL
+    @api_token = ENV.fetch("CODECLIMATE_API_TOKEN")
+  end
+
+  def request(path)
+    uri = URI("#{@base_uri}#{path}")
+    if ENV["DEBUG"] == "1"
+      STDERR.puts uri.to_s
+    end
+    http = Net::HTTP.new(uri.hostname, uri.port)
+    http.use_ssl = uri.scheme == "https" ? true : false
+    request = Net::HTTP::Get.new(uri.request_uri)
+    request.set_content_type("application/vnd.api+json")
+    request["Authorization"] = "Token token=#{@api_token}"
+    http.request(request)
+  end
+end
+
+class CodeClimate
+  def self.repo_stats(repo_name)
+    cc = CodeClimate.new
+    repo = cc.repo_from_github_slug(repo_name)
+    return {} if repo.nil?
+    
+    repo_id     = repo["id"]
+    snapshot_id = repo["relationships"]["latest_default_branch_snapshot"]["data"]["id"]
+
+    snapshot = cc.snapshot(repo_id, snapshot_id)
+    files    = cc.files(repo_id, snapshot_id)
+    issues   = cc.issues(repo_id, snapshot_id)
+
+    issue_counts = {}
+    issue_counts['total']       = issues.length
+    issue_counts['complexity']  = issues.count { |issue| issue["attributes"]["categories"].include?("Complexity") }
+    issue_counts['duplication'] = issues.count { |issue| issue["attributes"]["categories"].include?("Duplication") }
+    issue_counts['style']       = issues.count { |issue| issue["attributes"]["categories"].include?("Style") }
+    issue_counts['bug_risk']    = issues.count { |issue| issue["attributes"]["categories"].include?("Bug Risk") }
```

Comment:
Optional, but more_core_extensions has a helper for this called `element_counts`...Roughly something like the following I think would work...

```ruby
require "more_core_extensions/core_ext/array/element_counts"
issue_counts = issues.flat_map { |i| issue["attributes"]["categories"] }.element_counts
issue_counts['total'] = issues.length
```